packages/react-resizable-panels/src/PanelResizeHandleRegistry.ts

```diff
@@ -137,13 +156,39 @@ function recalculateIntersectingHandles({ x, y }: { x: number; y: number }) {
       ? hitAreaMargins.coarse
       : hitAreaMargins.fine;
 
-    const intersects =
+    const eventIntersects =
       x >= left - margin &&
       x <= right + margin &&
       y >= top - margin &&
       y <= bottom + margin;
 
-    if (intersects) {
+    if (eventIntersects) {
+      // TRICKY
+      // We listen for pointers events at the root in order to support hit area margins
+      // (determining when the pointer is close enough to an element to be considered a "hit")
+      // Clicking on an element "above" a handle (e.g. a modal) should prevent a hit though
+      // so at this point we need to compare stacking order of a potentially intersecting drag handle,
+      // and the element that was actually clicked/touched
+      if (
+        targetElement !== null &&
+        element !== targetElement &&
+        !element.contains(targetElement) &&
+        !targetElement.contains(element) &&
+        // Calculating stacking order has a cost, so we should avoid it if possible
+        // That is why we only check potentially intersecting handles,
+        // and why we skip if the event target is within the handle's DOM
+        compare(targetElement, element) > 0 &&
+        // If the target is above the drag handle, then we also need to confirm they overlap
+        // If they are beside each other (e.g. a panel and its drag handle) then the handle is still interactive
+        intersects(
+          targetElement.getBoundingClientRect(),
+          element.getBoundingClientRect(),
+          true
+        )
```

Comment:
Note to self:

It's not enough to compare the target here. The target might be a small element inside of a larger container. (For example, a SPAN or a DIV inside of a larger modal dialog).